data-prepper-plugins/kafka-plugins/src/integrationTest/java/org/opensearch/dataprepper/plugins/kafka/buffer/KafkaBufferOTelIT.java

```diff
@@ -0,0 +1,179 @@
+        when(topicConfig.getGroupId()).thenReturn("buffergroup-" + RandomStringUtils.randomAlphabetic(6));
+        when(topicConfig.isCreateTopic()).thenReturn(true);
+        when(topicConfig.getSerdeFormat()).thenReturn(MessageFormat.BYTES);
+        when(topicConfig.getWorkers()).thenReturn(1);
+        when(topicConfig.getMaxPollInterval()).thenReturn(Duration.ofSeconds(5));
+        when(topicConfig.getConsumerMaxPollRecords()).thenReturn(1);
+        when(topicConfig.getSessionTimeOut()).thenReturn(Duration.ofSeconds(15));
+        when(topicConfig.getHeartBeatInterval()).thenReturn(Duration.ofSeconds(3));
+        when(topicConfig.getAutoCommit()).thenReturn(false);
+        when(topicConfig.getAutoOffsetReset()).thenReturn("earliest");
+        when(topicConfig.getThreadWaitingTime()).thenReturn(Duration.ofSeconds(1));
+        when(kafkaBufferConfig.getTopic()).thenReturn(topicConfig);
+
+        EncryptionConfig encryptionConfig = mock(EncryptionConfig.class);
+
+        bootstrapServersCommaDelimited = System.getProperty("tests.kafka.bootstrap_servers");
+
+        LOG.info("Using Kafka bootstrap servers: {}", bootstrapServersCommaDelimited);
+
+        when(kafkaBufferConfig.getBootstrapServers()).thenReturn(Collections.singletonList(bootstrapServersCommaDelimited));
+        when(kafkaBufferConfig.getEncryptionConfig()).thenReturn(encryptionConfig);
+    }
+
+    private String getFileAsJsonString(String requestJsonFileName) throws IOException {
+        final StringBuilder jsonBuilder = new StringBuilder();
+        try (final InputStream inputStream = Objects.requireNonNull(
+                KafkaBufferOTelIT.class.getClassLoader().getResourceAsStream(requestJsonFileName))) {
+            final BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(inputStream));
+            bufferedReader.lines().forEach(jsonBuilder::append);
+        }
+        return jsonBuilder.toString();
+    }
+
+    private ExportMetricsServiceRequest buildExportMetricsServiceRequestFromJsonFile(String requestJsonFileName) throws IOException {
+        final ExportMetricsServiceRequest.Builder builder = ExportMetricsServiceRequest.newBuilder();
+        JsonFormat.parser().merge(getFileAsJsonString(requestJsonFileName), builder);
+        return builder.build();
+    }
+
+    @Test
+    void test_otel_metrics_with_kafka_buffer() throws Exception {
+        KafkaBuffer kafkaBuffer = new KafkaBuffer(pluginSetting, kafkaBufferConfig, pluginFactory, acknowledgementSetManager, pluginMetrics, new OTelMetricDecoder(), null, null);
+        buffer = new KafkaDelegatingBuffer(kafkaBuffer);
+        oTelMetricsGrpcService = new OTelMetricsGrpcService(10000, 
+            buffer,
+            pluginMetrics);
+
+        final ExportMetricsServiceRequest request = buildExportMetricsServiceRequestFromJsonFile(TEST_REQUEST_MULTIPLE_METRICS_FILE);
+        oTelMetricsGrpcService.rawExport(request);
```

Comment:
We don't need `rawExport ` method instead you can directly write to buffer

`buffer.writeBytes(request.toByteArray(), null, bufferWriteTimeoutInMillis);`

Reply:
Also, why are we not calling `export` method on oTelMetricsGrpcService ?

Reply:
Will change to either use writeBytes() directly or use `export` in the next PR.